### DISCUSSION TITLE
Add area sensor device class and unit conversions

### DISCUSSION THREAD

**Discussion by mikey0000** [home-assistant/architecture#1124] [Entity Models] [answered]
## Background

Several integrations are reporting area mowed or vacuumed, but lack the functionality to have the area converted according to the user's configured unit system. Integrations are currently mostly using AREA_SQUARE_METER (which has no conversion to US Customary).

## Proposal

Add an `area` sensor device class with support for unit conversion between US customary area measurements as well as metric area units

The following units should be supported:
```
SQUARE_METERS = "m²"
SQUARE_CENTIMETERS = "cm²"
SQUARE_KILOMETERS = "km²"
SQUARE_MILLIMETERS = "mm²"
SQUARE_INCHES = "in²"
SQUARE_FEET = "ft²"
SQUARE_YARDS = "yd²"
SQUARE_MILES = "mi²"
ACRES = "ac"
HECTARES = "ha"
```

Draft PR for reference: https://github.com/home-assistant/core/pull/123563/

### Integrations which can benefit from the proposal

Currently there are the following integrations making use of AREA_SQUARE_METER
mammotion (HACS robot lawn mower)/me
ecovacs
bloomsky
rituals_perfume_genie
roborock
romy
roomba
smartthings
xiaomi_miio

**Comment by emontnemery** [comment]
Please make the proposal more concrete:
- Which area units will be supported
- We'll also need a new sensor device class, please update the proposal with that

**Reply by mikey0000** [reply]
Updated, please advise if there is anything I've missed.

**Reply by emontnemery** [reply]
Thanks a lot! I did some adjustment of the formatting.

**Comment by emontnemery** [comment, marked as answer]
The proposal has been approved in the core architectural meeting ✅

**Reply by frenck** [reply]
Done in https://github.com/home-assistant/core/pull/123563

**Comment by make-all** [comment]
Perhaps there should be a checklist to ensure documentation is updated and announcements made about architectural changes that introduce deprecation warnings. Since the release of 2024.12 my issue tracker has been flooded with reports from users because HA is telling them to report a "bug" in my integration, but there is no mention of this in either the developer documentation, the user documentation or the developer blog (there is a one liner in the release notes, but it does not cover the scope of what has changed).

**Reply by mikey0000** [reply]
Seems the dev blog for it was not mergex
https://github.com/home-assistant/developers.home-assistant/pull/2448

**Reply by mikey0000** [reply]
I'd missed an important comment and will address it so it can be merged
